pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2019 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snapshot
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	k8errors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/kanisterio/kanister/pkg/kube/snapshot/apis/v1alpha1"
+	"github.com/kanisterio/kanister/pkg/poll"
+)
+
+const (
+	pvcKind = "PersistentVolumeClaim"
+
+	// Snapshot resource Kinds
+	VolSnapClassKind   = "VolumeSnapshotClass"
+	VolSnapKind        = "VolumeSnapshot"
+	VolSnapContentKind = "VolumeSnapshotContent"
+)
+
+var (
+	// VolSnapGVR specifies GVR schema for VolumeSnapshots
+	VolSnapGVR = schema.GroupVersionResource{Group: v1alpha1.GroupName, Version: v1alpha1.Version, Resource: v1alpha1.VolumeSnapshotResourcePlural}
+	// VolSnapClassGVR specifies GVR schema for VolumeSnapshotClasses
+	VolSnapClassGVR = schema.GroupVersionResource{Group: v1alpha1.GroupName, Version: v1alpha1.Version, Resource: v1alpha1.VolumeSnapshotClassResourcePlural}
+	// VolSnapContentGVR specifies GVR schema for VolumeSnapshotContents
+	VolSnapContentGVR = schema.GroupVersionResource{Group: v1alpha1.GroupName, Version: v1alpha1.Version, Resource: v1alpha1.VolumeSnapshotContentResourcePlural}
+)
+
+type SnapshotAlpha struct {
+	kubeCli kubernetes.Interface
+	dynCli  dynamic.Interface
+}
+
+func (sna *SnapshotAlpha) GetVolumeSnapshotClass(annotationKey, annotationValue string) (string, error) {
+	us, err := sna.dynCli.Resource(VolSnapClassGVR).Namespace("").List(metav1.ListOptions{})
+	if err != nil {
+		return "", errors.Errorf("Failed to get VolumeSnapshotClasses in the cluster: %v", err)
+	}
+	if us == nil || len(us.Items) == 0 {
+		return "", errors.Errorf("Failed to find any VolumeSnapshotClass in the cluster: %v", err)
+	}
+	for _, vsc := range us.Items {
+		ans := vsc.GetAnnotations()
+		if val, ok := ans[annotationKey]; ok && val == annotationValue {
+			return vsc.GetName(), nil
+		}
+	}
+	return "", errors.Errorf("Failed to find VolumesnapshotClass with %s annotation in the cluster", annotationKey)
+}
+
+// Create creates a VolumeSnapshot and returns it or any error happened meanwhile.
```

Comment:
error "that" happened

Reply:
@bathina2 for one word changes like this one, you can use github suggestions

```suggestion
// Create creates a VolumeSnapshot and returns it or any error that happened meanwhile.
```

Reply:
Yea I used it elsewhere below